### ISSUE TITLE
Narrator and NVDA is not reading out the fast-text-field component

### ISSUE THREAD

**Issue by khamudom** [microsoft/fast#3436]
**Describe the bug; what happened?**
While running Narrator or NVDA, as you tab onto the fast-text-field component, it is silent.

**What are the steps to reproduce the issue?**

Example:

1. Run Narrator or NVDA
2. Go to https://explore.fast.design/components/fast-text-field'
3. tab onto the text-field. Observe that the screen reader not saying anything.

**What behavior did you expect?**
Would expect the screen reader to say "edit"

**If applicable, provide screenshots:**

**In what environment did you see the issue?**
Windows 10
Microsft Edge browser

* OS & Device: [e.g. MacOS, iOS, Windows, Linux] on [iPhone 7, PC]
* Browser [e.g. Edge, Chrome, Safari, FireFox]
* Version [e.g. 1.8.0]

**Is there any additional context?**

**Comment by khamudom** [comment]
Looking at the template, I notice we are missing the `type` attribute.
I can add the `type` attribute to the template, but `role="textbox"` will have to be removed for assistive technologies to work, at least for Narrator and NVDA. Since this a single line input, just adding `type` should be enough for assistive technologies to notice it.

The suggested change to the template would be to remove the `role` and add `type`
```html
<template
        appearance="${x => x.appearance}"
        aria-required="${x => x.required}"
        aria-disabled="${x => x.disabled}"
        aria-readonly="${x => x.readOnly}"
        tabindex="${x => (x.disabled ? null : 0)}"
        type=${x => x.type}
        class="
            ${x => x.appearance}
            ${x => (x.readOnly ? "readonly" : "")}
        "
    >
```

**Comment by chrisdholt** [comment]
I don't think this is the issue at play here - I think there needs to be more investigation in how this relates to #3248.

**Comment by nicholasrice** [comment]
@khamudom I think you can pull *all* of the ARIA related attributes off of this component. This component delegates focus to the shadowed `<input>` element, so there is no need for the host to define itself as a textbox.

**Comment by chrisdholt** [comment]
@nicholasrice in the event we delegate focus to the shadowDOM does that fully defer anything on the host? This would explain issues related to #3248

**Comment by nicholasrice** [comment]
I believe so yes. [This explainer](https://github.com/TakayoshiKochi/tabindex-focus-navigation-explainer/blob/master/TabindexFocusNavigationExplainer.md) goes into significant detail on what `delegatesFocus` does. It is rather old (2015) so I'm unsure if changes to the behavior have been made.

An interesting finding from that document - elements implementing `delegatesFocus` don't need to add `[tabindex=0]` to the host element - `delegatesFocus` makes them focusable.

**Comment by chrisdholt** [comment]
Thanks for the link. I was aware of the second bit of information there, was interesting to find that to be the case when implementing :). 

I think adding the type to the input recieving focus will fix the issue here - as part of that, we can drop the aria attributes @khamudom.
